File: Codechef/starters 138/tempCodeRunnerFile.py
from sys import stdin, stdout

input = stdin.readline

def sieve(n):
    is_prime = [True] * (n + 1)
    p = 2
    while (p * p <= n):
        if is_prime[p]:
            for i in range(p * p, n + 1, p):
                is_prime[i] = False
        p += 1
    primes = [p for p in range(2, n + 1) if is_prime[p]]
    return primes

primes = sieve(10**5)
prime_squares = [p**2 for p in primes]
prime_squares_set = set(prime_squares)

def can_sum_to_n(n):
    for i in range(len(prime_squares)):
        for j in range(i + 1, len(prime_squares)):
            remainder = n - (prime_squares[i] + prime_squares[j])
            if remainder in prime_squares_set and remainder != prime_squares[i] and remainder != prime_squares[j]:
                return True
    return False

T = int(input().strip())
results = []
for _ in range(T):
    N = int(input().strip())
    if can_sum_to_n(N):
        results.append("YES")
    else:
        results.append("NO")

stdout.write("\n".join(results) + "\n")

File: Codechef/starters 138/tempCodeRunnerFile.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <unordered_set>

using namespace std;

vector<int> sieve(int n) {
    vector<bool> is_prime(n + 1, true);
    vector<int> primes;
    for (int p = 2; p * p <= n; ++p) {
        if (is_prime[p]) {
            for (int i = p * p; i <= n; i += p) {
                is_prime[i] = false;
            }
        }
    }
    for (int p = 2; p <= n; ++p) {
        if (is_prime[p]) {
            primes.push_back(p);
        }
    }
    return primes;
}

bool can_sum_to_n(int n, const vector<int>& prime_squares) {
    unordered_set<int> squares_set(prime_squares.begin(), prime_squares.end());
    int length = prime_squares.size();
    for (int i = 0; i < length; ++i) {
        for (int j = i + 1; j < length; ++j) {
            int required = n - (prime_squares[i] + prime_squares[j]);
            if (squares_set.find(required) != squares_set.end() && required != prime_squares[i] && required != prime_squares[j]) {
                return true;
            }
        }
    }
    return false;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int T;
    cin >> T;
    vector<int> primes = sieve(100000);
    vector<int> prime_squares;
    for (int prime : primes) {
        prime_squares.push_back(pow(prime, 2));
    }

    vector<string> results;
    for (int t = 0; t < T; ++t) {
        int N;
        cin >> N;
        if (can_sum_to_n(N, prime_squares)) {
            results.push_back("YES");
        } else {
            results.push_back("NO");
        }
    }

    for (const string& result : results) {
        cout << result << '\n';
    }

    return 0;
}
